### DISCUSSION TITLE
Option to have ignore patterns for `public-hoist-pattern` etc

### DISCUSSION THREAD

**Discussion by mjyocca** [pnpm/pnpm#4282] [Q&A] [answered]
I might be mistaken but not seeing any way to explicitly tell pnpm to ignore hoisting a package to the root node_modules.

```
# Defaults
public-hoist-pattern[]='*types*'
public-hoist-pattern[]='*eslint*'
public-hoist-pattern[]='@prettier/plugin-*'
public-hoist-pattern[]='*prettier-plugin-*'

#Ignore 
public-hoist-pattern[]='!types/package-to-ignore'
```

I might be using the wrong tool for the job here but thoughts?

**Comment by zkochan** [comment, marked as answer]
It is not very intuitive but if you set something to public-hoist-pattern then you remove the defaults:

```
public-hoist-pattern[]=
```

**Reply by mjyocca** [reply]
Right, 

this is probably an edge case, have a workspace project with conflicts between different versions of  `@aws-sdk/types` package, want to try and minimize changes for right now by allowing the current defaults but just excluding this package.

I was able to exclude this package by modifying the `*types*` to `*@types*`, but was curious if there was a way to explicitly not hoist a package to the root, even if it matches the patterns by default or added manually.

**Reply by zkochan** [reply]
We don't have an exclude syntax now for this setting.

**Reply by mjyocca** [reply]
No worries! have a good workaround for the time being.

**Reply by ConcernedHobbit** [reply]
@zkochan We're running into this as an issue as well, we need to have two versions of a same typing package and hoisting them both causes issues with tsc. Is exclude syntax for hoisting on the roadmap anywhere? Inverse excluding (explicitly including everything else) seems a bit too much.

**Reply by NexZhu** [reply]
+1 for this, need different TypeScript versions for a monorepo project

**Reply by zkochan** [reply]
Created an issue https://github.com/pnpm/pnpm/issues/5272
